nl_server/registry.py

```diff
@@ -53,14 +63,14 @@ def __init__(self, server_config: ServerConfig):
   def get_index(self, index_type: str) -> Embeddings:
     return self.name_to_emb.get(index_type)
 
-  def get_reranking_model(self, model_name: str) -> RerankingModel:
-    return self.name_to_rank_model.get(model_name)
-
   def get_attribute_model(self) -> AttributeModel:
     return self._attribute_model
 
-  def get_embedding_model(self, model_name: str) -> EmbeddingsModel:
-    return self.name_to_emb_model.get(model_name)
+  def get_model(
+      self,
+      model_name: str) -> tuple[(EmbeddingsModel | RerankingModel, ModelUsage)]:
+    return (self.name_to_model.get(model_name),
+            self._server_config.models.get(model_name).usage)
```

Comment:
Hmm, wonder why not keep the original API, but throw exception if get_reranking_model matches an embedding model and vice versa?

Reply:
yea, that's better.